kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/JdkTls.java

```diff
@@ -81,17 +97,50 @@ public SSLContext sslContext() {
         }
     }
 
-    @NonNull
-    private SSLContext customSSLContext(TrustProvider trust) {
-        TrustManager[] trustManagers = getTrustManagers(trust);
-        return getSslContext(trustManagers);
+    /* exposed for testing */
+    static KeyManager[] getKeyManagers(KeyProvider key) {
+        return key.accept(new KeyProviderVisitor<>() {
+            @Override
+            public KeyManager[] visit(KeyPair keyPair) {
+                throw new SslConfigurationException("KeyPair is not supported by vault KMS yet");
+            }
+
+            @Override
+            public KeyManager[] visit(io.kroxylicious.proxy.config.tls.KeyStore keyStore) {
+                try {
+                    if (keyStore.isPemType()) {
+                        throw new SslConfigurationException("PEM is not supported by vault KMS yet");
+                    }
+                    KeyStore store = KeyStore.getInstance(keyStore.getType());
+                    try (FileInputStream fileInputStream = new FileInputStream(keyStore.storeFile())) {
+                        char[] storePassword = passwordOrNull(keyStore.storePasswordProvider());
+                        store.load(fileInputStream, storePassword);
+                    }
+                    char[] keyPassword = passwordOrNull(keyStore.keyPasswordProvider());
```

Comment:
In io.kroxylicious.proxy.config.tls.NettyKeyProvider#keyManagerFactory if the keyPassword is null, we pass the keyStorePassword instead.    The intent was to support the common case where the two passwords were the same.